src/ext/java/org/opentripplanner/ext/flex/FlexRouter.java

```diff
@@ -0,0 +1,200 @@
+package org.opentripplanner.ext.flex;
+
+import gnu.trove.set.TIntSet;
+import org.opentripplanner.common.model.T2;
+import org.opentripplanner.ext.flex.distancecalculator.DirectDistanceCalculator;
+import org.opentripplanner.ext.flex.distancecalculator.DistanceCalculator;
+import org.opentripplanner.ext.flex.template.FlexAccessTemplate;
+import org.opentripplanner.ext.flex.template.FlexEgressTemplate;
+import org.opentripplanner.ext.flex.trip.FlexTrip;
+import org.opentripplanner.model.Stop;
+import org.opentripplanner.model.StopLocation;
+import org.opentripplanner.model.Transfer;
+import org.opentripplanner.model.calendar.ServiceDate;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.routing.algorithm.raptor.transit.StopIndexForRaptor;
+import org.opentripplanner.routing.algorithm.raptor.transit.mappers.DateMapper;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graphfinder.StopAtDistance;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class FlexRouter {
+
+  /* Transit data */
+  private final Graph graph;
+  private final Collection<StopAtDistance> streetAccesses;
+  private final Collection<StopAtDistance> streetEgresses;
+  private final StopIndexForRaptor stopIndex;
+  private final Collection<Transfer> transitTransfers;
+  private final FlexIndex flexIndex;
+  private final DistanceCalculator distanceCalculator;
+
+  /* Request data */
+  private final ZonedDateTime startOfTime;
+  private final int departureTime;
+  private final boolean arriveBy;
+
+  private final TIntSet[] servicesRunning;
+  private final int[] differenceFromStartOfTime;
+  private final ServiceDate[] serviceDates;
+
+  /* State */
+  private List<FlexAccessTemplate> flexAccessTemplates = null;
+  private List<FlexEgressTemplate> flexEgressTemplates = null;
+
+  public FlexRouter(
+      RoutingRequest request,
+      int additionalPastSearchDays,
+      int additionalFutureSearchDays,
+      Collection<StopAtDistance> streetAccesses,
+      Collection<StopAtDistance> egressTransfers,
+      StopIndexForRaptor stopIndex
```

Comment:
Remove this parameter, the stopIndex - it should not be necessary if the mapping to Raptor is moved out of flex.